### PR TITLE
reactify

### PR DESCRIPTION
**Pull request by EspenRoth**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,96 @@
...
+// import './App.css';
+import './css/bootstrap.css';
+import './css/bootstrap-grid.css';
+import './css/app.css';
+
+// spotify:track:0ROCe58wXm7sBUPkZaRrnY
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {errors: [], now_playing: null, queue: null, inputValue: 'spotify:track:0ROCe58wXm7sBUPkZaRrnY'};
```
Comment (craigcabrey) [inline src/App.js]: nah

File: src/App.js [hunk 2]
```diff
@@ -1,26 +1,96 @@
...
+  }
+  componentDidMount() {
+    this.state = {now_playing: null, queue: null};
+    fetch("http://127.0.0.1:5000/api/")
+    .then(res => {return res.json()})
+    .then( res => {
+      console.log(res);
```
Comment (craigcabrey) [inline src/App.js]: kill console logs plzzle

File: src/App.js [hunk 3]
```diff
@@ -1,26 +1,96 @@
...
+// spotify:track:0ROCe58wXm7sBUPkZaRrnY
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {errors: [], now_playing: null, queue: null, inputValue: 'spotify:track:0ROCe58wXm7sBUPkZaRrnY'};
+  }
+  componentDidMount() {
+    this.state = {now_playing: null, queue: null};
+    fetch("http://127.0.0.1:5000/api/")
+    .then(res => {return res.json()})
```
Comment (craigcabrey) [inline src/App.js:21]: there's gotta be someway to register this as a global response middleware handler

File: src/App.js [hunk 4]
```diff
@@ -1,26 +1,96 @@
...
+                     className="form-control"
+                     placeholder="Spotify URI, or URL (tracks or playlists)" 
+                     value={this.state.inputValue} onChange={(event) => { this.handleInputChange(event); }}/>
+              <div className="input-group-append">
+                  <button className="btn btn-success" onClick={(event) => { this.queueSong(event); }}>
+                      Queue
+                  </button>
+              </div>
+          </div>
+          {
```
Comment (craigcabrey) [inline src/App.js]: don't inline this, too much cancer

File: src/components/NowPlaying.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+
+class NowPlaying extends Component {
+	render () {
```
Comment (craigcabrey) [inline src/components/NowPlaying.js]: spaces, not tabs

File: src/components/NowPlaying.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+
+class NowPlaying extends Component {
+	render () { 
+		if(this.props.now_playing) {
```
Comment (craigcabrey) [inline src/components/NowPlaying.js]:
```suggestion
		if (this.props.now_playing) {
```

File: app.py [hunk 7]
```diff
@@ -10,15 +10,15 @@
 import threading
 
 import flask
+from flask_cors import CORS
 import spotipy
 import spotipy.oauth2
 import spotipy.util
 
 app = flask.Flask('spockify')
+CORS(app, resources=r'/api/.*', allow_headers='Content-Type')
```
Comment (craigcabrey) [inline app.py]: should probably only do this in development mode

File: app.py [hunk 8]
```diff
@@ -41,6 +41,70 @@ def human_duration(self):
...
+
+    if now_playing_track:
+        now_playing_track['artist'] = ', '.join(
+            artist['name'] for artist in now_playing_track['item']['artists']
+        )
+
+        for image in now_playing_track['item']['album']['images']:
+            if image['height'] == 300:
+                now_playing_track['album_art_url'] = image['url']
+    ret = {
```
Comment (craigcabrey) [inline app.py:56]: useless variable

File: src/App.js [hunk 9]
```diff
@@ -1,26 +1,96 @@
...
+      this.setState({now_playing: res.now_playing, queue: res.queue});
+    });
+  }
+
+  queueSong(e) {
+    e.preventDefault();
+    fetch('http://127.0.0.1:5000/api/', {
+      method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({spotify_uri: this.state.inputValue})
```
Comment (craigcabrey) [inline src/App.js:32]: do we have to do this? intuition tells me fetch() should do this automagically but not sure

File: src/App.js (hunk 1 above)
Comment (EspenRoth) [reply src/App.js]: wym, seperate lines? or a default track in the queue?

File: src/App.js (hunk 9 above)
Comment (EspenRoth) [reply src/App.js:32]: yes because post I believe we need it all - could experiment a bit

File: src/App.js (hunk 3 above)
Comment (EspenRoth) [reply src/App.js:21]: we can wrap it later if we want, return the promise from fetchJson or whatever we wanna call it

**Pull request merged**